### ISSUE TITLE
[BUG] Animating display from 'block' to 'none' not working since 11.0.11

### ISSUE THREAD

**Issue by dd-jonas** [framer/motion#2656]
**1. Read the [FAQs](#faqs) 👇**

**2. Describe the bug**

When defining variants with `display: block` and `display: none`, the display does not return to `none` upon triggering the animation.

**3. IMPORTANT: Provide a CodeSandbox reproduction of the bug**

[Reproduction](https://stackblitz.com/edit/vitejs-vite-mc7x2z?file=src%2FApp.tsx)

**4. Steps to reproduce**

1. Go to the sandbox above
2. Click on the show/hide button to show the overlay
3. Click on the show/hide button to hide the overlay
4. Notice that the opacity has animated, but display is still block (notice that the backdrop is still there, because you can't select nay text on the screen)

**5. Expected behavior**

The overlay should return to `display: none` as was the case before 11.0.11.

**6. Video or screenshots**

/

**7. Environment details**

/

**Comment by mattgperry** [comment]
Yeah this did get inadvertently broken though the previous behaviour wasn't great either (instantly animating to "none" when you probably want that to happen at the end of the animation)

I'll take a look over the next couple days and see if we can do something where animating `display` and `visibility` can be handled more gracefully.

**Comment by alex-cortenix** [comment]
Any progress on that??
